### PR TITLE
Prevent using `trigger_rule=TriggerRule.ALWAYS` in a task-generated mapping within bare tasks

### PR DESCRIPTION
**Pull request by shahar1**
related: #43368

Thanks to Jed's [comment](https://github.com/apache/airflow/pull/43368#issuecomment-2523903356) in the previous PR, I've realized that:
1. I've made a mistake in terminology, and mixed up **dynamic task mapping** with **task-generated mapping**.
2. The previous PR covered only the case of mapped task groups, but didn't cover the case of bare tasks (see example in the comment / test).

This PR fixes related docs and newsfragment of #43368, while applying the prevention for bare tasks.

---
**^ Add meaningful description above**
Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/contributing-docs/05_pull_requests.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

File: newsfragments/43368.improvement.rst [hunk 1]
```diff
@@ -0,0 +1 @@
+``TriggerRule.ALWAYS`` cannot be utilized within a task-generated mapping within mapped task groups
```
Comment (eladkal) [inline newsfragments/43368.improvement.rst]: This note mark as improvment but it feels more like a bug fix? Can we rephrase it to clarify the bug and the fix?

Comment (jedcunningham) [reply newsfragments/43368.improvement.rst]: We should likely just combine these into a single newfragment too. +1 on bugfix.

**Review by jedcunningham — COMMENTED** [review]
Overall, looks good! Thanks @shahar1!

File: newsfragments/43368.improvement.rst (hunk 1 above)
Comment (shahar1) [reply newsfragments/43368.improvement.rst]: Combined and renamed into a bug fix, calrified the bug and the fix

**Review by eladkal — APPROVED** [review]
LGTM

File: newsfragments/43368.improvement.rst (hunk 1 above)
Comment (shahar1) [reply newsfragments/43368.improvement.rst]: Had to squeeze it into one line because of the newsfragment pre-commit

**Review by jedcunningham — APPROVED** [review]

**Comment by shahar1** [comment]
Working on a backport :)
